### ISSUE TITLE
Resources directory written for non-action commands like help

### ISSUE THREAD

**Issue by coemgenuslcp** [pml-lang/pml-companion#87]
In PML 3.0.0, with the resources directory being written to `~/.config/PML_Companion/`, this happens even when you run non-action commands such as `pml -version`.
```
INFO: Creating shared data directory ?/.config/PML_Companion/3_0
PMLC 3.0.0 2022-08-19
```
I can maybe see why you might desire contextual information on the resources (including overrides) available for the help messages, but I can't imagine you would need the resources directory for getting the version.  In scripting, version checking is often a smoke test for checking a command-line tool is runnable, so the result here is that the check is not read-only.

**Comment by pml-lang** [comment]
In the current version the existence of the config directory is [checked](https://github.com/pml-lang/pml-companion/blob/15b18a7295d293bb6435a9fdf38ef6802843533c/pmlc-commands/src/main/java/dev/pmlc/commands/Start.java#L31) at each PMLC startup. It if doesn't exist it is created.

This should be changed. The directory should only be created when it is actually needed by a PMLC command.

**Comment by pml-lang** [comment]
> The directory should only be created when it is actually needed by a PMLC command.

Fixed in version 4.0.0
